### PR TITLE
Improve some kernel error messages

### PR DESCRIPTION
**Pull request by fingolfin**
This continues the work begun in PR #2947 and implements the change I promised there: Instead of "(not a integer)" we now say for example "(not the integer 5)".

Also adds and uses `RequireStringRep`, `RequireInt`, `RequireSmallInt`

### CODE REVIEW

**Comment by ChrisJefferson** [comment]
Picky, I don't really like the messages `(not true)` and `(not false)`, as in context it could be (I think) confusing, consider:

`<x> must be a small list (not true)` -- the `not true` could be referring to the fact the statement is false :) Therefore I'd prefer something else, perhaps `not the value 'true'`, or `not the boolean 'true'` (`value` has the advantage that `not the boolean `fail`, while a valid statement about GAP, also looks silly :) )

File: src/error.h [hunk 1]
```diff
@@ -145,15 +154,49 @@ ErrorReturnVoid(const Char * msg, Int arg1, Int arg2, const Char * msg2);
...
+        "must be an integer")
+
+
+/****************************************************************************
+**
+*F  RequireSmallInt
+*/
+#define RequireSmallInt(funcname, op, argname) \
+    RequireArgumentCondition(funcname, op, argname, IS_INTOBJ(op), \
+        "must be an integer")
```
Comment (ChrisJefferson) [inline src/error.h]: Should the refer to the "smallness" of the required integer somehow?

Comment (fingolfin) [reply src/error.h]: Ouch, indeed, that's a copy&past mistake. I'll change it to "must be a small integer"

**Comment by fingolfin** [comment]
How about `(not 'true')`?

**Pull request merged**